src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -1809,6 +1864,13 @@ private void parsePattern(String pattern) {
                     } else {
                         appendValue(new WeekBasedFieldPrinterParser(cur, count, count, 19));
                     }
+                } else if (cur == 'B') {
+                    switch (count) {
+                        case 1, 2, 3 -> appendDayPeriodText(TextStyle.SHORT);
```

Comment:
I think this should be `case 1`. The 2 and 3 are not in the Javadoc, and I don't think they are in LDML. I note that patterns G and E do this though, so there is precedent.

Reply:
Fixed.